### PR TITLE
SDIT-405: ✨ New prisoner numbers endpoint

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt [hunk 1]
```diff
@@ -193,6 +195,40 @@ class PrisonerResource(private val globalSearchService: GlobalSearchService) {
...
+  @SlowReportQuery
+  fun getPrisonerNumbers(
+    @RequestParam(value = "page", defaultValue = "0", required = false)
+    @Parameter(description = "The page number of the paged results")
+    page: Int,
+    @RequestParam(value = "size", defaultValue = "10", required = false)
+    @Parameter(description = "Requested limit to number of prisoner numbers returned.")
+    size: Int,
+  ): Page<String> {
+    val offenderNumbers = globalSearchService.getOffenderNumbers(page.toLong() * size, size.toLong())
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt:224]: note have to multiply page by size to get the offset

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/PrisonerResourceTest.kt [hunk 2]
```diff
@@ -145,12 +146,6 @@ class PrisonerResourceTest : ResourceTest() {
 
     @Test
     fun testCanReturnPrisonerInformationByNomsIdWhenReleased() {
-      webTestClient.get()
-        .uri("/api/prisoners/Z0023ZZ")
-        .headers(setAuthorisation(listOf("VIEW_PRISONER_DATA")))
-        .exchange()
-        .expectStatus().isOk
-        .expectBody().jsonPath("$[*].offenderNo", "Z0023ZZ")
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/PrisonerResourceTest.kt:153]: Can this test be deleted or is it to be replaced?

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/PrisonerResourceTest.kt [hunk 3]
```diff
@@ -180,4 +175,87 @@ class PrisonerResourceTest : ResourceTest() {
...
+        .jsonPath("content").value<List<String>> {
+          assertThat(it)
+            .hasSize(10)
+            .contains("A1181FF")
+            .doesNotContain("A1234AN", "A1234AO")
+        }
+    }
+
+    @Test
+    fun testReturn404WhenDoesNotHavePrivs() {
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/PrisonerResourceTest.kt:253]: minor typo in test name 404

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**